typescript/integrations.d.ts

```diff
@@ -22,90 +22,99 @@ type IntegrationBase = {
   createdAt: Date
 }
 
-export type IntegrationTrimEnvironment = {
+export type SavedSecret = {
+  cipherText: string
+  encryptionKeyId: string
+}
+export type NewSecret = string
+
+export type Secret<T> = T extends SavedSecret | NewSecret ? T : SavedSecret
```

Comment:
Why the generic? Can't this just be 

```suggestion
export type Secret = SavedSecret | NewSecret
```

Reply:
The idea is to give us control over which type we are using.

Reply:
Whilst minimising the need to actually pass generics to the types when using them. So you only need to pass anything if you need a `NewSecret`

Reply:
Having the `Secret` as a generic allows us to write

Reply:
Having the `Secret` as a generic allows us to write 
```
export type IntegrationTrimEnvironment<S = SavedSecret> = {
  id: string
  label: string
  baseUrl: string
  username: string
  password: Secret<S>
}
```

instead of 
```
export type IntegrationTrimEnvironment<S  extends Secret = SavedSecret> = {
  id: string
  label: string
  baseUrl: string
  username: string
  password: S
}
```

Just less code to write across every type dec. Easier to grasp IMO

Reply:
And it allows us to still use as 
```
IntegrationTrimEnvironment // password === SavedSecret
```
or
```
IntegrationTrimEnvironment<NewSecret> // password === NewSecret
```